### PR TITLE
Add a timeout for send/receive on the OMAPI socket

### LINKED ISSUE #1024: No timeout set when reading from OMAPI socket
There is not timeout set when reading from the OMAPI.

pfdhcplistener can wait indefinitly

### PR DESCRIPTION
**Pull request by julsemaan**
# Description

Added configurable timeout for send/receive operations on the OMAPI socket
# Impacts

OMAPI
# Issue

fixes #1024 
# Delete branch after merge

YES
# NEWS file entries
## Enhancements
- Added configurable timeout for send/receive operations on the OMAPI socket

### CODE REVIEW

**Comment by fdurand** [comment]
Can it be simplified by https://metacpan.org/pod/IO::Socket::Timeout ?

**Comment by julsemaan** [comment]
Could be, I will test it

**Comment by julsemaan** [comment]
Reworked, please re-review

**Comment by jrouzierinverse** [comment]
Update dependencies

**Comment by julsemaan** [comment]
What dependencies ?

**Comment by jrouzierinverse** [comment]
IO::Socket::Timeout

File: conf/pf.conf.defaults [hunk 1]
```diff
@@ -956,6 +956,11 @@ port=7911
 #
 # The OMAPI host
 host=localhost
+#
+# omapi.timeout
+#
+# The timeout on the OMAPI socket
+timeout=1
```
Comment (jrouzierinverse) [inline conf/pf.conf.defaults]: I think this is too long maybe 0.5 seconds would be better

**Comment by julsemaan** [comment]
No new package necessary for IO::Socket::Timeout

It is already provided by the packages we already install

File: conf/pf.conf.defaults (hunk 1 above)
Comment (julsemaan) [inline conf/pf.conf.defaults]: Agreed

File: lib/pf/OMAPI.pm [hunk 2]
```diff
@@ -241,6 +251,62 @@ sub connect {
...
+Validate the reply coming from the socket
+
+Return undef if invalid, return the reply if valid
+
+=cut
+
+sub validate_reply {
+    my ($self, $reply) = @_;
+    if (! $reply && ( 0+$! == ETIMEDOUT || 0+$! == EWOULDBLOCK )) {
+        pf::log::get_logger->error("Timeout while reading from the OMAPI socket");
```
Comment (jrouzierinverse) [inline lib/pf/OMAPI.pm]: use pf::log; On top

**Comment by julsemaan** [comment]
Fixed, re-review

**Comment by fdurand** [comment]
Tested and works as expected, merge

**Pull request merged**